lib/api/src/Cardano/Wallet/Api/Types/Error.hs

```diff
@@ -312,3 +315,10 @@ data ApiErrorNoSuchPool = ApiErrorNoSuchPool
     deriving (Data, Eq, Generic, Show, Typeable)
     deriving (FromJSON, ToJSON) via DefaultRecord ApiErrorNoSuchPool
     deriving anyclass NFData
+
+data ApiErrorNoSuchWallet = ApiErrorNoSuchWallet
+    { walletId :: !Text
```

Comment:
Suggestion: does it make sense to use `ApiT WalletId` here?

```suggestion
    { walletId :: !(ApiT WalletId)
```

Reply:
trying to use pattern synonyms here -> https://github.com/cardano-foundation/cardano-wallet/pull/4597/commits/984bb6eb3c841acaefb22251c42f58a0b7327462